src/aspire/estimation/covar2d.py

```diff
@@ -91,7 +93,7 @@ def _get_covar(self, coeffs, mean_coeff=None,  do_refl=True):
                 covar_coeff.append(covar_coeff_blk)
                 ind = ind + 1
 
-        return covar_coeff
+        return BlkDiagMatrix.from_blk_diag(covar_coeff, dtype=coeffs.dtype)
```

Comment:
The `from_blk_diag` call?  That is the constructor here.  I guess it would be possible to add methods to incrementally build up a BlockDiagonal, but as it is now, we need to know all the sizes apriori, one way or another.  To be incremental, that would require an `empty` and `append` method.  No problem, probably better. Let me know if you would rather see that in this request or a future smaller one.

Also, looking at this with fresh eyes, I am wondering if `from_list` is a better name now that `blk_diag_*` is removed from the python code. Could do that at same time...

Reply:
Gone.

Reply:
(I think) I originally meant that the `BlkDiagMatrix` would be built up incrementally. From what I understood, we don't need to know the sizes in advance, but maybe I missed something.

Reply:
Cool, it is incremental in the latest.